components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolverTest.java

```diff
@@ -0,0 +1,271 @@
+
+/**
+ * Test cases for IdentityKeyStoreResolver.
+ */
+public class IdentityKeyStoreResolverTest extends TestCase {
+
+    private static final String PRIMARY_KEY_STORE = "wso2carbon.jks";
+    private static final String PRIMARY_KEY_STORE_PASSWORD = "wso2carbon";
+    private static final String PRIMARY_KEY_STORE_ALIAS = "wso2carbon";
+
+    private static final String TENANT_KEY_STORE = "foo-com.jks";
+    private static final String TENANT_KEY_STORE_PASSWORD = "foo.com";
+    private static final String TENANT_KEY_STORE_ALIAS = "foo.com";
+
+    private static final String CUSTOM_KEY_STORE = "custom.jks";
+    private static final String CUSTOM_KEY_STORE_PASSWORD = "custom";
+    private static final String CUSTOM_KEY_STORE_ALIAS = "custom";
+
+    private static final String SUPER_TENANT_DOMAIN = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+    private static final String SUPER_TENANT_ID = "-1234";
+    private static final String TENANT_DOMAIN = "foo.com";
+    private static final String TENANT_ID = "1";
+
+    private KeyStore primaryKeyStore;
+    private KeyStore tenantKeyStore;
+    private KeyStore customKeyStore;
+
+    private static Key primaryKey;
+    private static Key tenantKey;
+    private static Key customKey;
+
+    private static Certificate primaryCertificate;
+    private static Certificate tenantCertificate;
+    private static Certificate customCertificate;
+
+    @Mock
+    private IdentityConfigParser mockIdentityConfigParser;
+
+    @Mock
+    private OMElement mockConfig;
+
+    private IdentityKeyStoreResolver identityKeyStoreResolver;
+
+    // Test key store mappings.
+    Map<InboundProtocol, IdentityKeyStoreMapping> keyStoreMappings = new ConcurrentHashMap<>();
+
+    private MockedStatic<IdentityConfigParser> identityConfigParser;
+    private MockedStatic<IdentityTenantUtil> identityTenantUtil;
+
+    @BeforeClass
+    public void setUp() throws Exception {
+
+        // Mock key store mapping configurations.
+        identityConfigParser = mockStatic(IdentityConfigParser.class);
+        mockIdentityConfigParser = mock(IdentityConfigParser.class);
+        identityConfigParser.when(IdentityConfigParser::getInstance).thenReturn(mockIdentityConfigParser);
+
+        mockConfig = mock(OMElement.class);
+        when(mockIdentityConfigParser.getConfigElement(anyString())).thenReturn(mockConfig);
+        when(mockConfig.getFirstChildWithName(any())).thenReturn(null);
+
+        IdentityKeyStoreMapping oauthKeyStoreMapping = new IdentityKeyStoreMapping(CUSTOM_KEY_STORE, InboundProtocol.OAUTH, true);
+        IdentityKeyStoreMapping wsTrustKeyStoreMapping = new IdentityKeyStoreMapping(CUSTOM_KEY_STORE, InboundProtocol.WS_TRUST, false);
+
+        keyStoreMappings.put(InboundProtocol.OAUTH, oauthKeyStoreMapping);
+        keyStoreMappings.put(InboundProtocol.WS_TRUST, wsTrustKeyStoreMapping);
+
+        setPrivateStaticField(IdentityKeyStoreResolver.class, "keyStoreMappings", keyStoreMappings);
```

Comment:
don't we have a setter method to set values for this attribute?

Reply:
Changed as discussed